### PR TITLE
SYS-3338 Converts Royalties and Batches size in bounded data

### PR DESCRIPTION
**Pull request by thadouk**
Converts the code to use bounded data for Royalties and Batches. Updates tests, adds new ones that test the boundaries and updates benchmarks code.
Removes older migration code as all chains have been migrated and there is no value converting the old data struct.

### CODE REVIEW

File: pallets/nft-manager/src/lib.rs [hunk 1]
```diff
@@ -112,6 +114,9 @@ pub mod pallet {
             + TypeInfo;
 
         type WeightInfo: WeightInfo;
+
+        #[pallet::constant]
```
Comment (nahuseyoum) [inline pallets/nft-manager/src/lib.rs:118]: :+1:

File: pallets/avn-proxy/src/tests/mock.rs [hunk 2]
```diff
@@ -113,6 +113,7 @@ impl pallet_nft_manager::Config for TestRuntime {
     type Public = AccountId;
     type Signature = Signature;
     type WeightInfo = ();
+    type BatchBound = ConstU32<10>;
```
Comment (nahuseyoum) [inline pallets/avn-proxy/src/tests/mock.rs:116]: I think you mean to use `BatchNftBound` here?

File: primitives/avn-common/src/lib.rs [hunk 3]
```diff
@@ -46,6 +46,8 @@ pub mod bounds {
     pub type MaximumValidatorsBound = ConstU32<256>;
     /// Bound used for voting session IDs
     pub type VotingSessionIdBound = ConstU32<64>;
+    /// Bound used for number of NFTs that can exist in a single Batch
+    pub type BatchNftBound = ConstU32<16384>;
```
[the remark this answers is not in the document]
Comment (thadouk) [reply primitives/avn-common/src/lib.rs:50]: Can do

File: pallets/avn-proxy/src/tests/mock.rs (hunk 2 above)
Comment (thadouk) [reply pallets/avn-proxy/src/tests/mock.rs:116]: We can use any bound we want for testing purposes since this is configurable.

**Review by nahuseyoum — APPROVED** [review]

**Review by ivan-cholakov — APPROVED** [review]

**Review by fluorostani — APPROVED** [review]

**Pull request merged**